packages/block-editor/src/hooks/style.js

```diff
@@ -98,28 +99,67 @@ export function getInlineStyles( styles = {} ) {
 	// The goal is to move everything to server side generated engine styles
 	// This is temporary as we absorb more and more styles into the engine.
 	const extraRules = getCSSRules( styles );
```

Comment:
`getCSSRules` will now generate the rules so that any `states` rules will have the psuedo selector as the `selector` prop of the rule.